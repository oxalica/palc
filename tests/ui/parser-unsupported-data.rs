use palc::Parser;

#[derive(Parser)]
struct Unit;

#[derive(Parser)]
struct Tuple();

#[derive(Parser)]
union Union {
    a: (),
}

// TODO: top-level subcommands.
#[derive(Parser)]
enum Enum {}

struct AssertImplParser
where
    Unit: Parser,
    Tuple: Parser,
    Enum: Parser;

fn main() {}
