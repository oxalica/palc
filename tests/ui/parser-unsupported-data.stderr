error: only structs with named fields are supported yet
 --> tests/ui/parser-unsupported-data.rs:3:10
  |
3 | #[derive(Parser)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `Parser` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only structs with named fields are supported yet
 --> tests/ui/parser-unsupported-data.rs:6:10
  |
6 | #[derive(Parser)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `Parser` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only structs with named fields are supported yet
 --> tests/ui/parser-unsupported-data.rs:9:10
  |
9 | #[derive(Parser)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `Parser` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only structs with named fields are supported yet
  --> tests/ui/parser-unsupported-data.rs:15:10
   |
15 | #[derive(Parser)]
   |          ^^^^^^
   |
   = note: this error originates in the derive macro `Parser` (in Nightly builds, run with -Z macro-backtrace for more info)
